contracts/ColonyFunding.sol

```diff
@@ -344,11 +344,12 @@ contract ColonyFunding is ColonyStorage, PatriciaTreeProofs { // ignore-swc-123
 
   function finalizeRewardPayout(uint256 _payoutId) public stoppable {
     RewardPayoutCycle memory payout = rewardPayoutCycles[_payoutId];
-
-    require(activeRewardPayouts[payout.tokenAddress], "colony-reward-payout-token-not-active");
+    require(payout.reputationState != 0x00, "colony-reward-payout-does-not-exist");
+    require(activeRewardPayouts[payout.tokenAddress] > 0, "colony-reward-payout-token-not-active");
+    require(_payoutId == activeRewardPayouts[payout.tokenAddress], "colony-reward-payout-not-most-recent");
```

Comment:
Alternatively, couldn't this function take `_token` instead, and then set `payoutId = activeRewardPayouts[_token]`? Seems like that would eliminate the need for some of this input validation.

Reply:
Yeah, but given that #754 is coming down the pipe very shortly which will have to use `_payoutId`, I didn't think I'd bother!

Reply:
Fair!